gpbft/messagetemplate.go

```diff
@@ -0,0 +1,97 @@
+package gpbft
+
+import (
+	"math/big"
+
+	xerrors "golang.org/x/xerrors"
+)
+
+type MessageTemplate struct {
+	powerTable powerTableAccess
```

Comment:
This feels a bit clumsy to me, though I see how it was reached from the current code. The power table came from the host in the first place, and we're just passing it back here. This field could instead be just an instance ID (if the host doesn't already know this), and then the host could pass the right power table to `Build`.

This is a symptom of another awkwardness, that the signing key is looked up by public key rather than by participant ID. Previously I thought this was not worth changing, but it may be contributing here.

Reply:
The point is that the MessageTemplate is exposed to whoever integrates it (Lotus), which can create the signing payload that can cross RPC boundaries to be signed and given back.
I will document it more, does that change your CR?

Reply:
I don't have a good enough view into how this will be used in the integration, so I won't hold you up on it. I do feel like we can do better, but don't know enough to say exactly how.